components/ledger/internal/storage/ledgerstore/balances.go

```diff
@@ -44,6 +46,18 @@ func (store *Store) GetAggregatedBalances(ctx context.Context, q *GetAggregatedB
 						default:
 							return "", nil, fmt.Errorf("unexpected type %T for column 'address'", address)
 						}
+					case metadataRegex.Match([]byte(key)):
+						if operator != "$match" {
+							return "", nil, errors.New("'metadata' column can only be used with $match")
+						}
+						match := metadataRegex.FindAllStringSubmatch(key, 3)
+						if !joinOnMetadataAdded {
```

Comment:
If joinOnMetadataAdded instead of if !joinOnMetadataAdded if I understand the name of the var ?